synchronizer/synchronizer.go

```diff
@@ -413,7 +413,7 @@ func (s *ClientSynchronizer) Sync() error {
 			// latestSequencedBatchNumber -> last batch on SMC
 			if latestSyncedBatch >= latestSequencedBatchNumber {
 				startTrusted := time.Now()
-				if s.syncTrustedStateExecutor != nil && !s.isTrustedSequencer {
+				if s.syncTrustedStateExecutor != nil && !s.isTrustedSequencer && s.cfg.L2Synchronization.Enable {
```

Comment:
Instead of adding a new condition in the if you can just leave s.syncTrustedStateExecutor as nil because it is not needed. Doing this, you will implement the requirement feature without wasting memory allocating something that it is not needed.